File: day02/day02-A.js
#!/usr/bin/env gjs

// Imports

imports.searchPath.unshift('../common');
const IO = imports.io;

// Functions

function process_commands(commands) {
    let [position, depth] = [0, 0];

    commands.forEach(command => {
    	let [direction, magnitude] = [command[0], parseInt(command[1])];

    	if (direction == 'forward') {
    	    position += magnitude;
	} else if (direction == 'down') {
    	    depth += magnitude;
	} else if (direction == 'up') {
    	    depth -= magnitude;
	}
    });

    return position * depth;
}

// Main Execution

let commands = IO.readlines(s => s.split(' '));
let result   = process_commands(commands);
IO.print(result);

File: day02/day02-A.py
#!/usr/bin/env python3

import sys

# Functions

def process_commands(commands):
    position, depth = 0, 0

    for direction, magnitude in commands:
        if direction == 'forward':
            position += int(magnitude)
        elif direction == 'down':
            depth += int(magnitude)
        elif direction == 'up':
            depth -= int(magnitude)

    return position, depth

# Main Execution

def main():
    commands        = [line.split() for line in sys.stdin]
    position, depth = process_commands(commands)
    print(position * depth)

if __name__ == '__main__':
    main()
